web/src/app/users/UserOnCallAssignmentList.tsx

```diff
@@ -56,22 +34,20 @@ interface Service {
 // with policy information mapped in the following structure:
 // {id: 'svc id', name: 'svc name', policyName: 'policy name', policySteps: [0,1,2]}
 //
-function services(onCallSteps: OnCallStep[] = []): Service[] {
+function services(onCallSteps: OnCallServiceAssignment[] = []): Service[] {
   const svcs: { [index: string]: Service } = {}
-  ;(onCallSteps || []).forEach((s) =>
-    (s.escalationPolicy.assignedTo || []).forEach((svc) => {
-      if (!svcs[svc.id]) {
-        svcs[svc.id] = {
-          id: svc.id,
-          name: svc.name,
-          policyName: s.escalationPolicy.name,
-          policySteps: [s.stepNumber],
-        }
-      } else {
-        svcs[svc.id].policySteps.push(s.stepNumber)
+  ;(onCallSteps || []).forEach((s) => {
```

Comment:
Not your code, but I imagine the ` || []` is not needed since we are defaulting to `[]` already in the function parameters

```suggestion
  ;(onCallSteps).forEach((s) => {
```

Reply:
If the backend returns null, the function defaults won't come into play. IIRC, those are only used if you call `services()` without an argument. As written, I don't think the default value would ever get used.

Reply:
Gotcha, I was under the impression `null` would trigger defaulting to `[]` in the params if present. Thanks, Javascript.